kvmagent/ansible/kvm.py

```
@@ -113,6 +113,169 @@ def update_libvirtd_config(host_post_info):
+        "  -CA {ca_dir}/cacert.pem -CAkey {ca_dir}/cakey.pem "
+        "  -CAcreateserial -out {tmp}/servercert.pem "
+        "  -extfile <(printf 'subjectAltName={san}') && "
+        "openssl genrsa -out {tmp}/clientkey.pem 4096 && "
+        "openssl req -new -key {tmp}/clientkey.pem "
+        "  -out {tmp}/client.csr -subj '/O=ZStack/CN={ip}' && "
+        "openssl x509 -req -days 3650 -in {tmp}/client.csr "
+        "  -CA {ca_dir}/cacert.pem -CAkey {ca_dir}/cakey.pem "
+        "  -CAcreateserial -out {tmp}/clientcert.pem "
+        "  -extfile <(printf 'subjectAltName={san}')"
+    ).format(tmp=cert_tmp_dir, ca_dir=ca_dir, ip=host_ip, san=san_entries)
+    shell_return = os.system("bash -c '%s'" % command.replace("'", "'\\''"))
```

Ruff (0.15.5)

[error] Starting a process with a shell, possible injection detected

(S605)